crates/sui-analytics-indexer/src/handlers/wrapped_object_handler.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::collections::BTreeMap;
+use std::path::Path;
+
+use sui_indexer::framework::Handler;
+use sui_package_resolver::Resolver;
+use sui_rest_api::{CheckpointData, CheckpointTransaction};
+use sui_types::object::Object;
+
+use crate::handlers::{get_move_struct, parse_struct, AnalyticsHandler};
+
+use crate::package_store::{LocalDBPackageStore, PackageCache};
+use crate::tables::WrappedObjectEntry;
+use crate::FileType;
+
+pub struct WrappedObjectHandler {
+    wrapped_objects: Vec<WrappedObjectEntry>,
+    package_store: LocalDBPackageStore,
+    resolver: Resolver<PackageCache>,
+}
+
+#[async_trait::async_trait]
+impl Handler for WrappedObjectHandler {
+    fn name(&self) -> &str {
+        "wrapped_object"
+    }
+    async fn process_checkpoint(&mut self, checkpoint_data: &CheckpointData) -> Result<()> {
+        let CheckpointData {
+            checkpoint_summary,
+            transactions: checkpoint_transactions,
+            ..
+        } = checkpoint_data;
+        for checkpoint_transaction in checkpoint_transactions {
+            for object in checkpoint_transaction.output_objects.iter() {
+                self.package_store.update(object)?;
```

Comment:
for my knowledge, does this have to be done before `process_transaction`, or it technically can be done in `process_transaction` together?

Reply:
I think you are probably correct in pointing this out that it could be done along with when we are processing the transaction. I had a thought about one transaction block containing multiple commands one of which publishes a package and another one which uses that package. I am almost certain it is not possible but just based on my lack of understanding I decided to update the package cache first.